### PR TITLE
Make `process()` functions `pub(crate)`

### LINKED ISSUE #338: Make internal `process()` functions `pub(crate)`
Similar to #324.

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #338

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics02_client/msgs/update_client.rs [hunk 1]
```diff
@@ -15,13 +15,14 @@ pub const TYPE_URL: &str = "/ibc.core.client.v1.MsgUpdateClient";
...
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct MsgUpdateClient {
+pub(crate) struct MsgUpdateClient {
     pub client_id: ClientId,
     pub header: Any,
     pub signer: Signer,
 }
 
 impl MsgUpdateClient {
+    #[allow(dead_code)]
```
Comment (plafer) [inline crates/ibc/src/core/ics02_client/msgs/update_client.rs]: We should just remove the constructor if it's unused. Same comment applies to all uses on `#[allow(dead_code)]`

File: crates/ibc/src/core/ics03_connection/msgs.rs [hunk 2]
```diff
@@ -24,11 +24,11 @@ pub mod conn_open_try;
 
 /// Enumeration of all possible messages that the ICS3 protocol processes.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum ConnectionMsg {
-    ConnectionOpenInit(MsgConnectionOpenInit),
-    ConnectionOpenTry(MsgConnectionOpenTry),
-    ConnectionOpenAck(MsgConnectionOpenAck),
-    ConnectionOpenConfirm(MsgConnectionOpenConfirm),
+pub(crate) enum ConnectionMsg {
+    Init(MsgConnectionOpenInit),
```
Comment (plafer) [inline crates/ibc/src/core/ics03_connection/msgs.rs]: Can we change all these to `OpenInit`, `OpenTry`, etc

**Review by plafer — CHANGES_REQUESTED** [review]
Thank you for handling this one @DaviRain-Su

File: crates/ibc/src/core/ics03_connection/msgs.rs (hunk 2 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics03_connection/msgs.rs]: Using `Open*` rust compiler clipy still holds errors.

**Comment by DaviRain-Su** [comment]
Because this `ValidationContext` trait is pub, it seems that `MsgEnvelope` must also be pub.
```rust
  pub trait ValidationContext {
        /// Validation entrypoint.
        fn validate(&self, message: MsgEnvelope) -> Result<(), RouterError>
        where
            Self: Sized,
        {
```

File: crates/ibc/src/core/ics26_routing/handler.rs [hunk 3]
```diff
@@ -44,7 +42,7 @@ where
 }
 
 /// Attempts to convert a message into a [MsgEnvelope] message
-pub fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
+pub(crate) fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
```
Comment (plafer) [inline crates/ibc/src/core/ics26_routing/handler.rs:45]: Let's leave this one `pub`, as there might be use cases for it (see [basecoin](https://github.com/informalsystems/basecoin-rs/blob/c5744f4a1eac9a63ef481410e52d9fb40363b97e/src/app/modules/ibc.rs#L298))

File: crates/ibc/src/core/ics26_routing/handler.rs [hunk 4]
```diff
@@ -53,12 +51,15 @@ pub fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
 /// and events produced after processing the input `msg`.
 /// If this method returns an error, the runtime is expected to rollback all state modifications to
 /// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
-pub fn dispatch<Ctx>(ctx: &mut Ctx, msg: MsgEnvelope) -> Result<HandlerOutput<()>, RouterError>
+pub(crate) fn dispatch<Ctx>(
```
Comment (plafer) [inline crates/ibc/src/core/ics26_routing/handler.rs:54]: Similarly, let's keep this one `pub` (see [basecoin](https://github.com/informalsystems/basecoin-rs/blob/c5744f4a1eac9a63ef481410e52d9fb40363b97e/src/app/modules/ibc.rs#L301))

**Review by plafer — COMMENTED** [review]
2 tiny changes left, and we're ready to merge

File: crates/ibc/src/core/ics02_client/msgs.rs [hunk 5]
```diff
@@ -14,6 +14,7 @@ pub mod misbehaviour;
 pub mod update_client;
 pub mod upgrade_client;
 
+#[allow(dead_code)]
```
Comment (plafer) [inline crates/ibc/src/core/ics02_client/msgs.rs:17]: I don't think we need this anymore, right?

File: crates/ibc/src/core/ics26_routing/handler.rs (hunk 4 above)
Comment (plafer) [reply crates/ibc/src/core/ics26_routing/handler.rs:54]: Cancel that, I'll post a new solution soon

File: crates/ibc/src/core/ics02_client/msgs.rs (hunk 5 above)
Comment (plafer) [reply crates/ibc/src/core/ics02_client/msgs.rs:17]: Cancel that, I'll post a new solution soon

File: crates/ibc/src/core/ics26_routing/handler.rs (hunk 3 above)
Comment (plafer) [reply crates/ibc/src/core/ics26_routing/handler.rs:45]: Cancel that, I'll post a new solution soon

File: .changelog/unreleased/improvements/324-make-msg-structs-pub-crate.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+- Make Msg* structs pub(crate) ([#324](https://github.com/cosmos/ibc-
```
Comment (plafer) [inline .changelog/unreleased/improvements/324-make-msg-structs-pub-crate.md:1]: I'll update this when merging #327.

**Review by plafer — APPROVED** [review]
Note that I made this PR close #338 instead. I'll fix #324 in an upcoming PR in light of learnings made in this PR.

Thank you!

**Pull request merged**